sklearn_numba_dpex/kmeans/kernels.py

```diff
@@ -466,7 +396,189 @@ def assignment(
         assignments_idx[sample_idx] = min_idx
         per_sample_inertia[sample_idx] = X_l2_norm + (two * min_sample_pseudo_inertia)
 
-    return assignment[global_size, work_group_size]
+    global_size = (math.ceil(n_samples / work_group_size)) * (work_group_size)
+    return (
+        n_centroids_private_copies,
+        fused_kmeans[global_size, work_group_size],
+        assignment[global_size, work_group_size],
+    )
+
+
+def _make_initialize_window_kernel_funcs(
+    n_clusters,
+    n_features,
+    work_group_size,
+    window_n_centroids,
+    window_n_features,
+    dtype,
+):
+    zero = dtype(0.0)
+    inf = dtype(math.inf)
+
+    @dpex.func
+    def _initialize_window_of_centroids(
+        local_work_id,
+        first_centroid_idx,
+        centroids_half_l2_norm,
+        centroids_window_half_l2_norm,
+        dot_products,
+    ):
+        # initialize the partial pseudo inertia for each of the window_n_centroids
+        # centroids in the window
+        for i in range(window_n_centroids):
+            dot_products[i] = zero
+
+        # the `window_n_centroids` first work items cooperate on loading the
+        # values of centroids_half_l2_norm relevant to current window. Each work
+        # item loads one single value.
+        half_l2_norm_loading_idx = first_centroid_idx + local_work_id
+        if local_work_id < window_n_centroids:
+            if half_l2_norm_loading_idx < n_clusters:
+                l2norm = centroids_half_l2_norm[half_l2_norm_loading_idx]
+            else:
+                l2norm = inf
+            centroids_window_half_l2_norm[local_work_id] = l2norm
+
+    n_window_features_per_work_group = work_group_size // window_n_centroids
+
+    centroids_window_height_ratio_multiplier = (
+        window_n_features // n_window_features_per_work_group
+    )
+
+    @dpex.func
+    def _load_window_of_centroids_and_features(
+        first_feature_idx,
+        loading_centroid_idx,
+        window_loading_centroid_idx,
+        window_loading_feature_offset,
+        current_centroids_t,
+        centroids_window,
```

Comment:
I think knowing which arguments is a tensor and if it gets modified really help both reading and understanding implementations. What do you think?
```suggestion
        current_centroids_t,    # IN
        centroids_window,       # OUT
```

The same suggestion apply to other kernels' signature.